### ISSUE TITLE
How to get 80% acc?

### ISSUE THREAD

**Issue by ShuangtongLi** [KaranPrajapati8750/CIFAR100-resnet9#1]
Hello! Thank for your work on fast implementation for cifar100 training. I ran experiment using your code, but I can only get 72% test acc. Did you use different hyperparameters in this experiment?

**Comment by KaranPrajapati8750** [comment]
Hi Chanderler , thanks for the appreciation. Please try to change the
values of the number of epochs, maximum learning rate, weight decay,
gradient clipping etc. Also try to uncomment some lines in the data
augmentation or randomization part or add some more randomizations. The
accuracy will surely increase.

**Comment by ShuangtongLi** [comment]
Thank you for your prompt reply. I tried several combinations of hyper perameters. Now the best one I can get is 74% acc. So what acc did you get when you did this experiment? And what about the hyperperamenters you used? Are the hyperperameters settings same as in the code? Looking forward to your reply!

**Comment by KaranPrajapati8750** [comment]
I'm running my code without changing anything but it's taking a lot of time
as google colab does not allow me to use GPU.. Will update you if it
finishes.

**Comment by ShuangtongLi** [comment]
Ok, thanks a lot!

**Comment by KaranPrajapati8750** [comment]
I ran it on my other id without changing anything and this is what I got.

**Comment by KaranPrajapati8750** [comment]
I trained the model for 20 epochs without changing anything and got 82%
accuracy in 10mins.

**Comment by ShuangtongLi** [comment]
Ok, I see where the problem is. Your code is only doing 20-way classification (You can print your classes and actually there are 20 classes instead of 100). The reason is that cifar-100 is consist of 20 hyper-classes. In your code, the torch.datasets.Imagefolder is used, and it will only consider the 20 hyperclasses as labels instead of the 100 ones. I ran your experiment and if doing 20-way classification task, the performance is 82% acc.

**Comment by KaranPrajapati8750** [comment]
Thanks for pointing that out. I'm a college student and complete beginner.
I worked with CIFAR10 first then changed the code for CIFAR100 that's why
it's predicting for 20 hyperclasses. Can you change that part in my code
.That'll be really helpful.
Thanks and regards.
